### DISCUSSION TITLE
What is the default boundary condition？

### DISCUSSION THREAD

**Discussion by Always-kimi** [idaholab/moose#27380] [Q&A General] [answered]
Hi there,
I've recently been learning to use the moose framework, and it‘s an excellent work. But I had some confusion when I learned about BCs block. There are some boundaries did not impose any condition, for example top and bottom boundaries in tutorials/darcy_thermo_mech/step01_diffusion/problems/step1.i 

```
[BCs]
  [inlet]
    type = DirichletBC  # Simple u=value BC
    variable = pressure # Variable to be set
    boundary = left     # Name of a sideset in the mesh
    value = 4000        # (Pa) From Figure 2 from paper.  First data point for 1mm spheres.
  []
  [outlet]
    type = DirichletBC
    variable = pressure
    boundary = right
    value = 0           # (Pa) Gives the correct pressure drop from Figure 2 for 1mm spheres
  []
[]
```

I like to know how moose handles these boundaries and if there are default boundary conditions？

**Comment by GiudGiud** [comment]
Hello

For a diffusion equation integrated by parts, the default boundary condition is 0 flux. 
The default boundary condition depends on the implementation of the equations.

Guillaume

**Reply by Always-kimi** [reply]
Hi Guillaume,

Thank you for your reply! So for each equation there are default boundary conditions, right? Do these boundary conditions depend on the kernels or on the computational feature?

Best regards,
Kimi

**Reply by GiudGiud** [reply, marked as answer]
The boundary conditions depend on the derivation of the weak form, and so do the kernels, so they are related and must be considered together

**Reply by Always-kimi** [reply]
Got it! Thanks so much!

**Reply by charlotte-b8086** [reply]
How do we know what the default boundary condition for an arbitrary kernel is? Do we simply assume it is the BC which sets any surface integrals in the weak form equal to 0?

Also, does a prescribed boundary condition always override a default boundary condition, or is it in conjunction to? For example, if you have a species diffusing with Dirichilet boundary conditions, do the Dirichilet boundary conditions override the homogeneous Neumann boundary condition, or are both boundary conditions active.

**Reply by GiudGiud** [reply]
> Do we simply assume it is the BC which sets any surface integrals in the weak form equal to 0?

basically. There are BCs that behave differently, for example Dirichlet boundary conditions which are different from IntegratedBCs and set the value of the variable (instead of contributing to the residual). 
EDIT: It doesnt set it to 0. it sets it the value of the BC.
You can also have penalty BCs on boundaries, which will they are summed are also applied due to the penalty the residual would suffer if they were not met. 

> Also, does a prescribed boundary condition always override a default boundary condition, or is it in conjunction to?

they are summed, just like kernels. Summing integrated BCs on top of a 0 default obviouslty replaces the 0. Same for penalty boundary conditons. 

> For example, if you have a species diffusing with Dirichilet boundary conditions, do the Dirichilet boundary conditions override the homogeneous Neumann boundary condition, or are both boundary conditions active.

Dirichlet BC actually override (well, depending on the discretization) because they impose the value of the variable directly.
MOOSE should error if you set inconsistent sets of boundary conditions, but it will depend on the discretization used, so do not rely on it

**Reply by charlotte-b8086** [reply]
Thank you for the clarification.
